napari/utils/events/dataclass.py

```diff
@@ -92,7 +93,7 @@ def set_with_events(self: C, name: str, value: Any) -> None:
     # TODO: use np.all(old_val == new_val)
 
     after = getattr(self, name)
-    if before != after:
+    if np.any(before != after):
```

Comment:
Is this the best approach? It makes the dataclass depend on numpy which somehow doesn't feel right, but isn't the worst thing in the world. Curious if @jni has ideas?

Reply:
it's the only thing that gets us a one-liner. If you want to compare iterables in Python, you'd do `if any(i != j for i, j in zip(before, after))`, but you'd first have to check that both were iterables. Seems painful...

Reply:
`np.any` works for me ... i agree that more thorough checking is painful.  We _might_ want to confirm performance here (since this will be called every time any attribute of any class is changed, right?)

Reply:
ok, will try and look into performance (i can try and compare both solutions there)

Reply:
i've added a check to see if either of these are np arrays and if so do the np.any call, so it should only happen now when needed